CONTRIBUTING.md

```diff
@@ -121,3 +122,120 @@ run it before submitting your PR. It runs `gofmt -s` (simplify) and `golint`.
 
 The dependencies are managed with `go mod` if you work with the sources under your
 `$GOPATH` you need to set the environment variable `GO111MODULE=on`.
+
+## Coding Guidelines
+
+Although OpenTelemetry project as a whole is still in Alpha stage we consider
+OpenTelemetry Collector to be close to production quality and the quality bar
+for contributions is set accordingly. Contributions must have readable code written
+with maintainability in mind (if in doubt check [Effective Go](https://golang.org/doc/effective_go.html)
+for coding advice). The code must adhere to the following robustness principles that
+are important for software that runs autonomously and continuously without direct
+interaction with a human (such as this Collector).
+
+### Startup Error Handling
+
+Verify configuration during startup and fail fast if the configuration is invalid.
+This will bring the attention of a human to the problem as it is more typical for humans
+to notice problems when the process is starting as opposed to problems that may arise
+sometime (potentially long time) after process startup. Monitoring systems are likely
+to automatically flag processes that exit with failure during startup, making it
+easier to notice the problem. The Collector should print a reasonable log message to
+explain the problem and exit with a non-zero code. It is acceptable to crash the process
+during startup if there is no good way to exit cleanly but do your best to log and
+exit cleanly with a process exit code.
+
+### Do not Crash after Startup
+
+Do not crash or exit the Collector process after the startup sequence is finished.
+A running Collector typically contains data that is received but not yet exported further
+(e.g. is stored in the queues and other processors). Crashing or exiting the Collector
+process will result in losing this data since typically the receiver has
+already acknowledged the receipt for this data and the senders of the data will
+not send that data again. This recommendation may change in the future if we
```

Comment:
Because of multiple pipelines and different possible configurations I don't think we should change this recommendation even in the presence of persistent queues. That said it already says "may change in the future" :) so no change required.

Reply:
Agreed. Actually the sentence is unnecessary, I will remove.